File: TimeSheetDPSidLib/ErrorLogger.vb
Imports Microsoft.VisualBasic
Imports System.IO

Public Class ErrorLogger 
    Dim _DestFolder As String
    Sub New(ByVal DestFolder As String)
        _DestFolder = DestFolder
        Try
            If Not Directory.Exists(DestFolder) Then Directory.CreateDirectory(DestFolder)
        Catch ex As Exception
        End Try
    End Sub
    Public Sub WriteClientLog(ByVal ex As String)

        Dim file As String = Path.Combine(_DestFolder, Now.Date.ToString("dd_MM_yyyy_") & "ClientErrorLog.txt")
        Dim hasInnerException As Boolean = False
        Using writer As New StreamWriter(file, True)
            writer.WriteLine(String.Format("======================================={0}=======================================", DateTime.Now.ToString()))
            writer.WriteLine(ex)
        End Using

    End Sub
    Public Sub WriteLog(ByVal ex As Exception)
        If ex.Message = "Thread was being aborted." Then
            Return
        End If

        Dim file As String = Path.Combine(_DestFolder, Now.Date.ToString("dd_MM_yyyy_") & "ErrorLog.txt")
        Dim hasInnerException As Boolean = False
        Using writer As New StreamWriter(file, True)
            writer.WriteLine(String.Format("======================================={0}=======================================", DateTime.Now.ToString()))
            writer.WriteLine(ex.Message)
            writer.WriteLine(String.Format("---------------Stack Trace---------------"))
            writer.WriteLine(ex.StackTrace)

            If ex.InnerException IsNot Nothing Then
                hasInnerException = True
                writer.WriteLine(String.Format("**********************Inner Exception**********************"))
            End If
        End Using

        If hasInnerException Then
            WriteLog(ex.InnerException)
        End If
    End Sub
    Public Sub WriteActivity(ByVal Activity As String)

        Dim file As String = Path.Combine(_DestFolder, Now.Date.ToString("dd_MM_yyyy_") & "ActivityLog.txt")
        Dim hasInnerException As Boolean = False
        Using writer As New StreamWriter(file, True)
            writer.WriteLine(String.Format("======================================={0}=======================================", DateTime.Now.ToString()))
            writer.WriteLine(Activity)
        End Using
    End Sub
End Class

File: AttendanceSystem/BaseClass/ErrorLogger.cs
using System;
using System.IO;
using System.Windows.Forms;

namespace AttendanceSystem.BaseClass
{

    public static class ErrorLoggerBase
    {
        public static ErrorLogger logger = new ErrorLogger(Application.StartupPath + "\\BASErrorLogs");
    }
    public class ErrorLogger
    {
        private string _DestFolder;
        public ErrorLogger(string DestFolder)
        {
            _DestFolder = DestFolder;
            try
            {
                if (!Directory.Exists(DestFolder))
                    Directory.CreateDirectory(DestFolder);
            }
            catch (Exception ex)
            {
            }
        }
        public void WriteClientLog(string ex)
        {
            string file = Path.Combine(_DestFolder, DateTime.Now.Date.ToString("dd_MM_yyyy_") + "ClientErrorLog.txt");
            bool hasInnerException = false;
            using (StreamWriter writer = new StreamWriter(file, true))
            {
                writer.WriteLine(string.Format("======================================={0}=======================================", DateTime.Now.ToString()));
                writer.WriteLine(ex);
            }
        }
        public void WriteLog(Exception ex)
        {
            if (ex.Message == "Thread was being aborted.")
                return;

            string file = Path.Combine(_DestFolder, DateTime.Now.Date.ToString("dd_MM_yyyy_") + "ErrorLog.txt");
            bool hasInnerException = false;
            using (StreamWriter writer = new StreamWriter(file, true))
            {
                writer.WriteLine(string.Format("======================================={0}=======================================", DateTime.Now.ToString()));
                writer.WriteLine(ex.Message);
                writer.WriteLine(string.Format("---------------Stack Trace---------------"));
                writer.WriteLine(ex.StackTrace);

                if (ex.InnerException != null)
                {
                    hasInnerException = true;
                    writer.WriteLine(string.Format("**********************Inner Exception**********************"));
                }
            }

            if (hasInnerException)
                WriteLog(ex.InnerException);
        }
        public void WriteActivity(string Activity)
        {
            string file = Path.Combine(_DestFolder, DateTime.Now.Date.ToString("dd_MM_yyyy_") + "ActivityLog.txt");
            bool hasInnerException = false;
            using (StreamWriter writer = new StreamWriter(file, true))
            {
                writer.WriteLine(string.Format("======================================={0}=======================================", DateTime.Now.ToString()));
                writer.WriteLine(Activity);
            }
        }
    }

}
